app/src/main/java/de/tum/in/tumcampusapp/component/ui/barrierfree/BarrierFreeFacilitiesActivity.kt

```diff
@@ -36,6 +38,16 @@ class BarrierFreeFacilitiesActivity : ActivityForAccessingTumCabe<List<RoomFinde
         binding = ActivityBarrierFreeFacilitiesBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
+        val toolbar = findViewById<Toolbar>(R.id.toolbar)
+        setSupportActionBar(toolbar)
+        supportActionBar?.let {
+            val parent = NavUtils.getParentActivityName(this)
+            if (parent != null) {
+                it.setDisplayHomeAsUpEnabled(true)
+                it.setHomeButtonEnabled(true)
+            }
+        }
```

Comment:
Can you please add a comment why this is needed? It seems like "busy code" that we used to have copied into every and all activities. Did you follow some documentation that points to this being required?

Reply:
Originally, this code is in the BaseActivity class, which therefore should normally inherited by the BarrierFreeFacilitiesActivity.
I debugged it for ~1hour and tested every possible change of layout files / etc..
The code in BaseActivity is run, though the back button is not enabled. I guess this is because the layout is inflated inside of the BarrierFreeFacilitiesActivity after super.onCreate.
I can check again but this was the only workaround getting the back button to work.